File: NatToInt.hs
nattoint :: Int -> Int
nattoint n
    | mod n 2 == 0 = 1 * (n `div` 2)
    | otherwise = -(n+1) `div` 2

nattoint' :: Int -> Int
nattoint' n = flip n * (n + oddoreven n) `div` 2
			  where flip x = truncate $ (-1)**(fromIntegral x);
			  		oddoreven x =  (`div` 2) $ (+1) $ flip(x-1)

check :: Int-> Bool
check n = map nattoint [0..n] == map nattoint [0..n]

File: NatToInt.jl
function nattoint(n)
	if (n % 2) == 0 
		return (-1 * (n/2))
    else
    	return ((n+1)/2)
    end
 end

function flip(x)
	return (-1)^float(x)
end

function ntoz(x)
	oddoreven = (flip(x-1) + 1)/2 # returns 1 for odd numbers and 0 for even numbers
	return flip(x) * (x+oddoreven(x))/2
end
